rpc_util.go

```diff
@@ -608,14 +612,17 @@ type parser struct {
 //   - an error from the status package
 //
 // No other error values or types must be returned, which also means
-// that the underlying io.Reader must not return an incompatible
+// that the underlying streamReader must not return an incompatible
 // error.
-func (p *parser) recvMsg(maxReceiveMessageSize int) (pf payloadFormat, msg []byte, err error) {
-	if _, err := p.r.Read(p.header[:]); err != nil {
+func (p *parser) recvMsg(maxReceiveMessageSize int) (payloadFormat, mem.BufferSlice, error) {
+	header, err := p.r.Read(5)
+	if err != nil {
 		return 0, nil, err
 	}
+	header.CopyTo(p.header[:])
```

Comment:
Shouldn't be necessary. We know it can't copy more than 5 bytes since `p.header` is an array, and `p.r.Read(5)` would have returned an error if the read was short.